spec/controllers/users_controller_spec.rb

```diff
@@ -112,7 +112,20 @@
         end
 
         it 'should save the user' do
-          expect(User.find(assigns(:user))).not_to be_nil
+          expect(User.find(assigns(:user).id)).not_to be_nil
+        end
+      end
+
+      context 'with CAS enabled' do
+        around(:example) do |example|
+          env_wrapper('CAS_AUTH' => '1') { example.run }
+        end
+
```

Comment:
So this is how we trick RSpec into modifying environment variables / "enabling CAS authentication". If you look in `spec/support/env_helpers.rb` you'll see the `env_wrapper` method definition. Basically what we're saying here is that for each test example (i.e. each `it` block), instead of simply running it we're passing that example to the `env_wrapper` method which spoofs a given environment variable for the duration of that example (so when our controller checks `ENV['CAS_AUTH']` it doesn't return `nil`).

Let me know if that makes sense!